[Bug]: Unintended cascade of workflow runs

## Expected behavior
When we publish a release on github, the following actions should run in the following order:
1. `next-release-minor.yml`
2. `populate-releasenotes.yml`
2. `deploy-docs.yml`

## Describe the bug
The following cascade of workflows ran when we published 0.4.0:
1. `deploy-docs.yml` Triggered via release: @[yardasol](https://github.com/yardasol) published [v0.4.0](https://github.com/arfc/saltproc/releases/tag/v0.4.0). Passed
2. `next-release-minor.yml` triggered via release: @[yardasol](https://github.com/yardasol) published [v0.4.0](https://github.com/arfc/saltproc/releases/tag/v0.4.0). Failed:
```
 HEAD detached from v0.4.0
nothing to commit, working tree clean
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>

Error: Process completed with exit code 128.
```
3. `test-saltproc.yml` Triggered via push: @[yardasol](https://github.com/yardasol) pushed c6b3d60 [v0.4.0](https://github.com/arfc/saltproc/tree/v0.4.0). Passed
4. `deploy-docs.yml` Triggered via release: @[yardasol](https://github.com/yardasol) edited [v0.4.0](https://github.com/arfc/saltproc/releases/tag/v0.4.0). Failed:
```
 /usr/bin/git push origin gh-pages
  To https://github.com/arfc/saltproc.git
   ! [remote rejected] gh-pages -> gh-pages (cannot lock ref 'refs/heads/gh-pages': is at 51e6cb39791df2755dc5045a0c61fb5192ea4307 but expected 5207c91751db83dfe4fddebf821392c6156258ba)
  error: failed to push some refs to 'https://github.com/arfc/saltproc.git'
  Error: Action failed with "The process '/usr/bin/git' failed with exit code 1"
```
5. `deploy-docs.yml` Triggered via release: @[yardasol](https://github.com/yardasol) edited [v0.4.0](https://github.com/arfc/saltproc/releases/tag/v0.4.0). PASSED
6. `populate-releasenotes.yml` Triggered via push: @[yardasol](https://github.com/yardasol) pushed c6b3d60 [v0.4.0](https://github.com/arfc/saltproc/tree/v0.4.0). FAILED:
```
Run pandoc -s -o RELEASENOTES.md -f rst -t gfm doc/releasenotes/v0.5.0.rst --columns 1000
pandoc: doc/releasenotes/v0.5.0.rst: openBinaryFile: does not exist (No such file or directory)
Error: Process completed with exit code 1.
```
7. `next-release-minor.yml` Triggered via releaseL @[yardasol](https://github.com/yardasol) edited [v0.4.0](https://github.com/arfc/saltproc/releases/tag/v0.4.0). FAILED:
```
Run gh api repos/arfc/saltproc/milestones \
  gh api repos/arfc/saltproc/milestones \
  -H "Authorize: token ***" \
  -H "Accept: application/vnd.github.v3+json" \
  -X POST \
  -F title=v0.5.0 
  shell: /usr/bin/bash -l {0}
  env:
    RELEASE_VERSION: v0.4.0
    NEW_RELEASE_VERSION: v0.5.0
    GITHUB_TOKEN: ***
gh: Validation Failed (HTTP 422)
{"message":"Validation Failed","errors":[{"resource":"Milestone","code":"already_exists","field":"title"}],"documentation_url":"https://docs.github.com/rest/reference/issues#create-a-milestone"}
Error: Process completed with exit code 1.
```
8. `next-release-minor.yml` Triggered via release: @[yardasol](https://github.com/yardasol) released [v0.4.0](https://github.com/arfc/saltproc/releases/tag/v0.4.0). FAILED:
```
Run gh api repos/arfc/saltproc/milestones \
  gh api repos/arfc/saltproc/milestones \
  -H "Authorize: token ***" \
  -H "Accept: application/vnd.github.v3+json" \
  -X POST \
  -F title=v0.5.0 
  shell: /usr/bin/bash -l {0}
  env:
    RELEASE_VERSION: v0.4.0
    NEW_RELEASE_VERSION: v0.5.0
    GITHUB_TOKEN: ***
gh: Validation Failed (HTTP 422)
{"message":"Validation Failed","errors":[{"resource":"Milestone","code":"already_exists","field":"title"}],"documentation_url":"https://docs.github.com/rest/reference/issues#create-a-milestone"}
Error: Process completed with exit code 1.
```
9. `deploy-docs.yml` Triggered via release: @[yardasol](https://github.com/yardasol) released [v0.4.0](https://github.com/arfc/saltproc/releases/tag/v0.4.0). FAILED:
```
Push the commit or tag
  /usr/bin/git push origin gh-pages
  To https://github.com/arfc/saltproc.git
   ! [rejected]        gh-pages -> gh-pages (fetch first)
  error: failed to push some refs to 'https://github.com/arfc/saltproc.git'
  hint: Updates were rejected because the remote contains work that you do
  hint: not have locally. This is usually caused by another repository pushing
  hint: to the same ref. You may want to first integrate the remote changes
  hint: (e.g., 'git pull ...') before pushing again.
  hint: See the 'Note about fast-forwards' in 'git push --help' for details.
  Error: Action failed with "The process '/usr/bin/git' failed with exit code 1"
```

## To Reproduce
Steps to reproduce the behavior:
1. Don't do this

## System Specifications:
 <!-- At a minimum, include the SaltProc version and versions of software directly related to the bug-->
 - Software version: 0.4.0
 - Subsystem: Github Actions: `ubuntu-latest` runner

## How can this issue be closed?
Some points of note:
1. The first run of `deploy-docs.yml` was triggered by publishing 0.4.0. **The `published` trigger should be removed from this workflow**, since `next-release-minor.yml` makes changes that will trigger `deploy-docs.yml` anyways.
2. The first run of `next-release-minor.yml` failed due to **improper git commands** (specifically, not pushing to HEAD)
3. The 2nd and 3rd run of `deploy-docs.yml` occurred at the same time, so there was a race-condition that caused one of them to fail when pushing to `gh-pages`. They were triggered by editing the v0.4.0 release body (this is a `published` webhook event, so it will trigger the workflow). Again, another reason to remove that trigger from `deploy-docs.yml`
4.  The first run of `populate-releasenotes.yml` failed because the `v0.5.0.rst` file that was supposed to be created by `next-release-minor.yml` was never created due to that run failing. This workflow was triggered by the creation of the v0.4.0 tag. **We should modify the triggers for this action so that it only runs from activity on the master branch**
5. The 2nd and 3rd run of `next-release-minor.yml` both failed due to the v0.5.0 milestone already being created in the first place. These workflows should not have been triggered in the first place. One of the webhooks was `released` and another was `edited`. **We should specify that this workflow only runs when a release is actually `released`**

--- a/.github/workflows/deploy-docs.yml
+++ b/.github/workflows/deploy-docs.yml
@@ -3,8 +3,6 @@ name: Deploy SaltProc docs
 
 on:
   push:
-    branches:
-      - 'master'
     paths:
       - 'doc/**'
       - 'saltproc/**'

--- a/.github/workflows/deploy-docs.yml
+++ b/.github/workflows/deploy-docs.yml
@@ -3,6 +3,8 @@ name: Deploy SaltProc docs
 
 on:
   push:
+    branches:
+      - 'master'
     paths:
       - 'doc/**'
       - 'saltproc/**'

--- a/.github/workflows/deploy-docs.yml
+++ b/.github/workflows/deploy-docs.yml
@@ -2,8 +2,6 @@
 name: Deploy SaltProc docs
 
 on:
-  release:
-    type: [published]
   push:
     paths:
       - 'doc/**'

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -68,15 +68,15 @@ jobs:
           sed -i "s/vx.x.x/${{ env.NEW_RELEASE_VERSION }}/g" doc/releasenotes/${{ env.NEW_RELEASE_VERSION }}.rst
           sed -i "s/${{ env.RELEASE_VERSION }}/${{ env.NEW_RELEASE_VERSION }}\n  ${{ env.RELEASE_VERSION }}/g" doc/releasenotes/index.rst
           echo "Make edits to version.py"
-          sed -i "s/_version_micro = '*\d*'*/_version_micro = ''/g" saltproc/version.py
-          sed -i "s/_version_minor = \d*/_version_minor = ${{ env.MINOR_VERSION }}/g" saltproc/version.py
+          sed -i "s/_version_micro = '*[0-9]*'*/_version_micro = ''/g" saltproc/version.py
+          sed -i "s/_version_minor = [0-9]*/_version_minor = ${{ env.MINOR_VERSION }}/g" saltproc/version.py
           sed -i "s/^# _version_extra = 'dev'/_version_extra = 'dev'/g" saltproc/version.py
           sed -i "s/^_version_extra = ''/# _version_extra = ''/g" saltproc/version.py
           echo "Add, commit, and push changes"
           git add doc/releasenotes/${{ env.RELEASE_VERSION }}.rst
           git add doc/releasenotes/${{ env.NEW_RELEASE_VERSION }}.rst
           git add doc/releasenotes/index.rst
-          git commit -am "created ${{ env.NEW_RELEASE_VERSION }} release notes"
+          git commit -m "created ${{ env.NEW_RELEASE_VERSION }} release notes"
           git add saltproc/version.py 
-          git commit -am "updated version to ${{ env.NEW_RELEASE_VERSION }}-dev"
+          git commit -m "updated version to ${{ env.NEW_RELEASE_VERSION }}-dev"
           git push origin HEAD:master

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -3,7 +3,7 @@ name: Create next minor release
 
 on:
   release:
-    type: [published]
+    type: [released]
   # enable worflow to be run manually
   workflow_dispatch:
 

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -53,7 +53,6 @@ jobs:
             
 
       - name: Create a new release
-        if: !${{ env.RELEASE_EXISTS }}
         run: |
           gh api repos/${{ github.repository }}/releases \
           -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
@@ -62,6 +61,8 @@ jobs:
           -F name=${{ env.NEW_RELEASE_VERSION }} \
           -F tag_name=${{ env.NEW_RELEASE_VERSION }} \
           -F draft=true 
+
+        if: !${{ env.RELEASE_EXISTS }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
@@ -84,13 +85,13 @@ jobs:
 
 
       - name: Create a new milestone.
-        if: !${{ env.MILESTONE_EXISTS }}
         run: |
           gh api repos/${{ github.repository }}/milestones \
           -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
           -H "Accept: application/vnd.github.v3+json" \
           -X POST \
           -F title=${{ env.NEW_RELEASE_VERSION }} 
+        if: !${{ env.MILESTONE_EXISTS }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -78,4 +78,4 @@ jobs:
           git commit -am "created ${{ env.NEW_RELEASE_VERSION }} release notes"
           git add saltproc/version.py 
           git commit -am "updated version to ${{ env.NEW_RELEASE_VERSION }}-dev"
-          git push
+          git push origin HEAD:master

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -68,8 +68,8 @@ jobs:
           sed -i "s/vx.x.x/${{ env.NEW_RELEASE_VERSION }}/g" doc/releasenotes/${{ env.NEW_RELEASE_VERSION }}.rst
           sed -i "s/${{ env.RELEASE_VERSION }}/${{ env.NEW_RELEASE_VERSION }}\n  ${{ env.RELEASE_VERSION }}/g" doc/releasenotes/index.rst
           echo "Make edits to version.py"
-          sed -i "s/_version_micro = *'*[0-9]*'/_version_micro = ''/g" saltproc/version.py
-          sed -i "s/_version_minor = *'*[0-9]*'/_version_minor = ${{ env.MINOR_VERSION }}/g" saltproc/version.py
+          sed -i "s/_version_micro = '*\d*'*/_version_micro = ''/g" saltproc/version.py
+          sed -i "s/_version_minor = \d*/_version_minor = ${{ env.MINOR_VERSION }}/g" saltproc/version.py
           sed -i "s/^# _version_extra = 'dev'/_version_extra = 'dev'/g" saltproc/version.py
           sed -i "s/^_version_extra = ''/# _version_extra = ''/g" saltproc/version.py
           echo "Add, commit, and push changes"

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -40,29 +40,19 @@ jobs:
           -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
           -H "Accept: application/vnd.github.v3+json" \
           -X GET \
-          --jq ''.[] | .name')
+          --jq '.[] | .name')
           if [[ $RELEASES =~ ${{ env.NEW_RELEASE_VERSION }} ]]
           then
-            EXISTS='true'
+            echo "Release exists"
           else
-            EXISTS='false'
+            gh api repos/${{ github.repository }}/releases \
+            -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            -X POST \
+            -F name=${{ env.NEW_RELEASE_VERSION }} \
+            -F tag_name=${{ env.NEW_RELEASE_VERSION }} \
+            -F draft=true 
           fi
-          echo "RELEASE_EXISTS="$EXISTS >> $GITHUB_ENV
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-            
-
-      - name: Create a new release
-        run: |
-          gh api repos/${{ github.repository }}/releases \
-          -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
-          -H "Accept: application/vnd.github.v3+json" \
-          -X POST \
-          -F name=${{ env.NEW_RELEASE_VERSION }} \
-          -F tag_name=${{ env.NEW_RELEASE_VERSION }} \
-          -F draft=true 
-
-        if: !(${{ env.RELEASE_EXISTS }})
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
@@ -72,26 +62,17 @@ jobs:
           -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
           -H "Accept: application/vnd.github.v3+json" \
           -X GET \
-          --jq ''.[] | .title')
+          --jq '.[] | .title')
           if [[ $MILESTONES =~ ${{ env.NEW_RELEASE_VERSION }} ]]
           then
-            EXISTS='true'
+            echo "Milestone exists"
           else
-            EXISTS='false'
+            gh api repos/${{ github.repository }}/milestones \
+            -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            -X POST \
+            -F title=${{ env.NEW_RELEASE_VERSION }} 
           fi
-          echo "MILESTONE_EXISTS="$EXISTS >> $GITHUB_ENV
-        env: 
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-
-      - name: Create a new milestone.
-        run: |
-          gh api repos/${{ github.repository }}/milestones \
-          -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
-          -H "Accept: application/vnd.github.v3+json" \
-          -X POST \
-          -F title=${{ env.NEW_RELEASE_VERSION }} 
-        if: !(${{ env.MILESTONE_EXISTS }})
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -31,6 +31,7 @@ jobs:
           MINOR_VERSION=$((${arr[1]}+1))
           arr[1]=$MINOR_VERSION
           NEW_RELEASE_VERSION="${arr[0]}.${arr[1]}.${arr[2]}"
+          echo "MINOR_VERSION="$MINOR_VERSION >> $GITHUB_ENV
           echo "NEW_RELEASE_VERSION="$NEW_RELEASE_VERSION >> $GITHUB_ENV
 
       - name: Create a new release
@@ -70,7 +71,7 @@ jobs:
           sed -i "s/_version_micro = *'*[0-9]*'/_version_micro = ''/g" saltproc/version.py
           sed -i "s/_version_minor = *'*[0-9]*'/_version_minor = ${{ env.MINOR_VERSION }}/g" saltproc/version.py
           sed -i "s/^# _version_extra = 'dev'/_version_extra = 'dev'/g" saltproc/version.py
-          sed -i "s/^_version_extra = ''# _version_extra = ''/g" saltproc/version.py
+          sed -i "s/^_version_extra = ''/# _version_extra = ''/g" saltproc/version.py
           echo "Add, commit, and push changes"
           git add doc/releasenotes/${{ env.RELEASE_VERSION }}.rst
           git add doc/releasenotes/${{ env.NEW_RELEASE_VERSION }}.rst

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -62,7 +62,7 @@ jobs:
           -F tag_name=${{ env.NEW_RELEASE_VERSION }} \
           -F draft=true 
 
-        if: !${{ env.RELEASE_EXISTS }}
+        if: !(${{ env.RELEASE_EXISTS }})
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
@@ -91,7 +91,7 @@ jobs:
           -H "Accept: application/vnd.github.v3+json" \
           -X POST \
           -F title=${{ env.NEW_RELEASE_VERSION }} 
-        if: !${{ env.MILESTONE_EXISTS }}
+        if: !(${{ env.MILESTONE_EXISTS }})
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -34,7 +34,26 @@ jobs:
           echo "MINOR_VERSION="$MINOR_VERSION >> $GITHUB_ENV
           echo "NEW_RELEASE_VERSION="$NEW_RELEASE_VERSION >> $GITHUB_ENV
 
+      - name: Check for existing release
+        run: |
+          RELEASES=$(gh api repos/${{ github.repository }}/releases \
+          -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
+          -H "Accept: application/vnd.github.v3+json" \
+          -X GET \
+          --jq ''.[] | .name')
+          if [[ $RELEASES =~ ${{ env.NEW_RELEASE_VERSION }} ]]
+          then
+            EXISTS='true'
+          else
+            EXISTS='false'
+          fi
+          echo "RELEASE_EXISTS="$EXISTS >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            
+
       - name: Create a new release
+        if: !${{ env.RELEASE_EXISTS }}
         run: |
           gh api repos/${{ github.repository }}/releases \
           -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
@@ -46,8 +65,26 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Check for existing milestone
+        run: |
+          MILESTONES=$(gh api repos/${{ github.repository }}/milestones \
+          -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
+          -H "Accept: application/vnd.github.v3+json" \
+          -X GET \
+          --jq ''.[] | .title')
+          if [[ $MILESTONES =~ ${{ env.NEW_RELEASE_VERSION }} ]]
+          then
+            EXISTS='true'
+          else
+            EXISTS='false'
+          fi
+          echo "MILESTONE_EXISTS="$EXISTS >> $GITHUB_ENV
+        env: 
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 
       - name: Create a new milestone.
+        if: !${{ env.MILESTONE_EXISTS }}
         run: |
           gh api repos/${{ github.repository }}/milestones \
           -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
@@ -56,6 +93,22 @@ jobs:
           -F title=${{ env.NEW_RELEASE_VERSION }} 
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Update version.py
+        run: |
+          echo "Configure git"
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          echo "Make edits to version.py"
+          sed -i "s/_version_micro = '*[0-9]*'*/_version_micro = ''/g" saltproc/version.py
+          sed -i "s/_version_minor = [0-9]*/_version_minor = ${{ env.MINOR_VERSION }}/g" saltproc/version.py
+          sed -i "s/^# _version_extra = 'dev'/_version_extra = 'dev'/g" saltproc/version.py
+          sed -i "s/^_version_extra = '0'/# _version_extra = '0'/g" saltproc/version.py
+          echo "Add, commit, and push changes"
+          git add saltproc/version.py 
+          git commit -m "updated version to ${{ env.NEW_RELEASE_VERSION }}-dev"
+          git push origin HEAD:master
+
 
       - name: Create new release notes
         run: |
@@ -67,16 +120,9 @@ jobs:
           cp doc/releasenotes/template.rst doc/releasenotes/${{ env.NEW_RELEASE_VERSION }}.rst
           sed -i "s/vx.x.x/${{ env.NEW_RELEASE_VERSION }}/g" doc/releasenotes/${{ env.NEW_RELEASE_VERSION }}.rst
           sed -i "s/${{ env.RELEASE_VERSION }}/${{ env.NEW_RELEASE_VERSION }}\n  ${{ env.RELEASE_VERSION }}/g" doc/releasenotes/index.rst
-          echo "Make edits to version.py"
-          sed -i "s/_version_micro = '*[0-9]*'*/_version_micro = ''/g" saltproc/version.py
-          sed -i "s/_version_minor = [0-9]*/_version_minor = ${{ env.MINOR_VERSION }}/g" saltproc/version.py
-          sed -i "s/^# _version_extra = 'dev'/_version_extra = 'dev'/g" saltproc/version.py
-          sed -i "s/^_version_extra = '0'/# _version_extra = '0'/g" saltproc/version.py
           echo "Add, commit, and push changes"
           git add doc/releasenotes/${{ env.RELEASE_VERSION }}.rst
           git add doc/releasenotes/${{ env.NEW_RELEASE_VERSION }}.rst
           git add doc/releasenotes/index.rst
           git commit -m "created ${{ env.NEW_RELEASE_VERSION }} release notes"
-          git add saltproc/version.py 
-          git commit -m "updated version to ${{ env.NEW_RELEASE_VERSION }}-dev"
           git push origin HEAD:master

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -3,7 +3,7 @@ name: Create next minor release
 
 on:
   release:
-    type: [released]
+    types: [released]
   # enable worflow to be run manually
   workflow_dispatch:
 

--- a/.github/workflows/next-release-minor.yml
+++ b/.github/workflows/next-release-minor.yml
@@ -71,7 +71,7 @@ jobs:
           sed -i "s/_version_micro = '*[0-9]*'*/_version_micro = ''/g" saltproc/version.py
           sed -i "s/_version_minor = [0-9]*/_version_minor = ${{ env.MINOR_VERSION }}/g" saltproc/version.py
           sed -i "s/^# _version_extra = 'dev'/_version_extra = 'dev'/g" saltproc/version.py
-          sed -i "s/^_version_extra = ''/# _version_extra = ''/g" saltproc/version.py
+          sed -i "s/^_version_extra = '0'/# _version_extra = '0'/g" saltproc/version.py
           echo "Add, commit, and push changes"
           git add doc/releasenotes/${{ env.RELEASE_VERSION }}.rst
           git add doc/releasenotes/${{ env.NEW_RELEASE_VERSION }}.rst

--- a/.github/workflows/populate-releasenotes.yml
+++ b/.github/workflows/populate-releasenotes.yml
@@ -42,7 +42,9 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Convert .rst to .md 
-        run: pandoc -o RELEASENOTES.md -f rst -t gfm doc/releasenotes/${{ env.RELEASE_VERSION }}.rst --columns 1000
+        run: |
+          pandoc -o RELEASENOTES.md -f rst -t gfm doc/releasenotes/${{ env.RELEASE_VERSION }}.rst --columns 1000
+          sed -i "s/# Release notes for ${{ env.RELEASE_VERSION }}\n//g" RELEASENOTES.md
 
 
       - name: Populate the release description with RELEASENOTES.md

--- a/.github/workflows/populate-releasenotes.yml
+++ b/.github/workflows/populate-releasenotes.yml
@@ -49,10 +49,16 @@ jobs:
 
       - name: Populate the release description with RELEASENOTES.md
         run: |
+          CURRENT_TAG=$(gh api repos/${{ github.repository }}/releases/${{ env.RELEASE_ID }} \
+          -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
+          -H "Accept: application/vnd.github.v3+json" \
+          -X GET \
+          --jq '.tag_name')
           gh api repos/${{ github.repository }}/releases/${{ env.RELEASE_ID }} \
           -H "Authorize: token ${{ secrets.GITHUB_TOKEN }}" \
           -H "Accept: application/vnd.github.v3+json" \
           -X PATCH \
+          -F tag_name=$CURRENT_TAG \
           -F body="$(cat RELEASENOTES.md)"
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/populate-releasenotes.yml
+++ b/.github/workflows/populate-releasenotes.yml
@@ -3,6 +3,8 @@ name: Populate SaltProc release notes
 
 on:
   push:
+    branches:
+      - 'master'
     paths:
       - 'doc/releasenotes/v**.rst'
   # enable worflow to be run manually

--- a/.github/workflows/populate-releasenotes.yml
+++ b/.github/workflows/populate-releasenotes.yml
@@ -42,7 +42,7 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Convert .rst to .md 
-        run: pandoc -s -o RELEASENOTES.md -f rst -t gfm doc/releasenotes/${{ env.RELEASE_VERSION }}.rst --columns 1000
+        run: pandoc -o RELEASENOTES.md -f rst -t gfm doc/releasenotes/${{ env.RELEASE_VERSION }}.rst --columns 1000
 
 
       - name: Populate the release description with RELEASENOTES.md

--- a/.github/workflows/populate-releasenotes.yml
+++ b/.github/workflows/populate-releasenotes.yml
@@ -3,8 +3,6 @@ name: Populate SaltProc release notes
 
 on:
   push:
-    branches:
-      - 'master'
     paths:
       - 'doc/releasenotes/v**.rst'
   # enable worflow to be run manually

--- a/.github/workflows/populate-releasenotes.yml
+++ b/.github/workflows/populate-releasenotes.yml
@@ -44,7 +44,7 @@ jobs:
       - name: Convert .rst to .md 
         run: |
           pandoc -o RELEASENOTES.md -f rst -t gfm doc/releasenotes/${{ env.RELEASE_VERSION }}.rst --columns 1000
-          sed -i "s/# Release notes for ${{ env.RELEASE_VERSION }}\n//g" RELEASENOTES.md
+          sed -i "s/# Release notes for ${{ env.RELEASE_VERSION }}//g" RELEASENOTES.md
 
 
       - name: Populate the release description with RELEASENOTES.md

